new-log-viewer/src/typings/worker.ts

```diff
@@ -76,6 +92,11 @@ type WorkerReqMap = {
         cursor: CursorType,
         decoderOptions?: DecoderOptionsType
     },
+    [WORKER_REQ_CODE.SET_FILTER]: {
+        cursor: CursorType,
+        logLevelFilter: LogLevelFilter,
+        decoderOptions?: DecoderOptionsType
```

Comment:
Why do we need the decoder options when setting a filter?

Reply:
I'm not sure. @junhaoliao, any idea why we have it for load page?

Reply:
The LOAD_PAGE requests carry a decoder option because we want to support changing the options without reloading the page. Previously, setting any decoder option was not so costly (like updating a format string, changing page size, etc.. In fact I see that we have renamed those concepts as "format" options in the decoders already). However, how the settings modal is implemented currently is that after setting any decoder option, the whole application gets reloaded, which makes this decoder option field in the LOAD_PAGE requests a bit unnecessary.

For now, I think it's fine to set the decoder options only in LOAD_FILE. If in the future we want to support setting those options without reloading the whole app, a new request code would be more appropriate.